gpytorch/lazy/added_diag_lazy_tensor.py

```diff
@@ -125,3 +130,23 @@ def _init_cache_for_non_constant_diag(self, eye, batch_shape, n):
         logdet = self._r_cache.diagonal(dim1=-1, dim2=-2).abs().log().sum(-1).mul(2)
         logdet -= (1.0 / self._noise).log().sum([-1, -2])
         self._precond_logdet_cache = logdet.view(*batch_shape) if len(batch_shape) else logdet.squeeze()
+
+    @cached(name="svd")
+    def _svd(self) -> Tuple["LazyTensor", Tensor, "LazyTensor"]:
+        if isinstance(self._diag_tensor, ConstantDiagLazyTensor):
+            U, S_, V = self._lazy_tensor.svd()
+            S = S_ + self._diag_tensor.diag()  # this assumes all diagonal entries are positive
+            return U, S, V
+        return super()._svd()
+
+    @cached(name="symeig")
+    def _symeig(self, eigenvectors: bool = False) -> Tuple[Tensor, Optional[LazyTensor]]:
+        try:
+            return pop_from_cache(self, "symeig", eigenvectors=True)
+        except CachingError:
+            pass
+        if isinstance(self._diag_tensor, ConstantDiagLazyTensor):
+            evals_, evecs = self._lazy_tensor.symeig(eigenvectors=eigenvectors)
+            evals = evals_ + self._diag_tensor.diag()  # this assumes all diagonal entries are positive
```

Comment:
nit: not sure why this assumes all diagonal entries are positive. If v is an eigenvector of A, then `(A - s I)v = Av - s*Iv = \lambda*v - s*v = (\lambda - s)*v`.

Reply:
that's a typo - this was meant to be "the same" rather than "positive". But I guess it's kid of moot now that we have an explicit `ConstantDiagLazyTensor` for that.